test/aqua/test_amplitude_estimation_circuitfactory.py

```diff
@@ -0,0 +1,623 @@
+
+        # qasm simulator
+        shots = 100
+        alpha = 0.01
+        result = qae.run(self._qasm(shots))
+        for method, expected_confint in expect.items():
+            confint = qae.confidence_interval(alpha, method)
+            self.assertEqual(confint, expected_confint)
+            self.assertTrue(confint[0] <= result[key] <= confint[1])
+
+    def test_iqae_confidence_intervals(self):
+        """End-to-end test for the IQAE confidence interval."""
+        n = 3
+        qae = IterativeAmplitudeEstimation(0.1, 0.01, a_factory=SineIntegralAFactory(n))
+        expected_confint = [0.19840508760087738, 0.35110155403424115]
+
+        # statevector simulator
+        result = qae.run(self._statevector)
+        confint = result['confidence_interval']
+        # confidence interval based on statevector should be empty, as we are sure of the result
+        self.assertAlmostEqual(confint[1] - confint[0], 0.0)
+        self.assertAlmostEqual(confint[0], result['estimation'])
+
+        # qasm simulator
+        shots = 100
+        result = qae.run(self._qasm(shots))
+        confint = result['confidence_interval']
+        self.assertEqual(confint, expected_confint)
+        self.assertTrue(confint[0] <= result['estimation'] <= confint[1])
+
+
+@ddt
+class TestCreditRiskAnalysis(QiskitAquaTestCase):
+    """Test a more difficult example, motived from Credit Risk Analysis."""
+
+    def setUp(self):
+        super().setUp()
+        warnings.filterwarnings(action="ignore", category=DeprecationWarning)
+
+    def tearDown(self):
+        super().tearDown()
+        warnings.filterwarnings(action="always", category=DeprecationWarning)
+
+    @data('statevector_simulator')
+    def test_conditional_value_at_risk(self, simulator):
+        """ conditional value at risk test """
+        # define backend to be used
+        backend = BasicAer.get_backend(simulator)
+
+        # set problem parameters
+        n_z = 2
+        z_max = 2
+        # z_values = np.linspace(-z_max, z_max, 2 ** n_z)
+        p_zeros = [0.15, 0.25]
+        rhos = [0.1, 0.05]
+        lgd = [1, 2]
+        k_l = len(p_zeros)
+        # alpha = 0.05
+
+        # set var value
+        var = 2
+        var_prob = 0.961940
+
+        # determine number of qubits required to represent total loss
+        # n_s = WeightedSumOperator.get_required_sum_qubits(lgd)
+
+        # create circuit factory (add Z qubits with weight/loss 0)
+        agg = WeightedSumOperator(n_z + k_l, [0] * n_z + lgd)
+
+        # define linear objective
+        breakpoints = [0, var]
+        slopes = [0, 1]
+        offsets = [0, 0]  # subtract VaR and add it later to the estimate
+        f_min = 0
+        f_max = 3 - var
+        c_approx = 0.25
+
+        # construct circuit factory for uncertainty model (Gaussian Conditional Independence model)
+        gci = GCI(n_z, z_max, p_zeros, rhos)
+
+        cvar_objective = PwlObjective(
```

Comment:
I cannot recall whether I added this test case to get test coverage of this and/or the MultivariateProblem. I recall we had no unit testing of some of these components so borrowed some notebook logic to build out some tests. If this is moved to some circuitfactory specific test that is intended to be dropped later we will need to check that we do not lose coverage of these.

Reply:
Yes, I'm also looking into the MultivariateProblem itself after this is done so we shouldn't lose coverage of it 👍